### PR TITLE
create reminder request

### PR DESCRIPTION
**Pull request by jitendrajaria**
request for creating the reminder

### CODE REVIEW

**Review by aayushKhandpur — CHANGES_REQUESTED** [review]
Address review

File: src/main/java/creativei/entity/Reminder.java [hunk 1]
```diff
@@ -1,18 +1,31 @@
...
+import java.text.ParseException;
 import java.util.Date;
 
 @Entity
 public class Reminder extends BaseEntity {
     private String title;
     private String description;
     private Date reminderTime;
+    @ManyToOne
     private Inquiry inquiry;
```
Comment (aayushKhandpur) [inline src/main/java/creativei/entity/Reminder.java:18]: Inquiry should be optional per inquiry

Comment (jitendrajaria) [reply src/main/java/creativei/entity/Reminder.java:18]: sir i don't get it. should i have to change the relationship?

File: src/main/java/util/LocalizationUtil.java [hunk 2]
```diff
@@ -16,4 +16,14 @@ public static final Date stringToDateConverter(String string) throws ParseExcept
             return null;
         return dateFormat.parse(string);
     }
+    static SimpleDateFormat simpleDateFormat=new SimpleDateFormat("yyyy-MM-dd hh:mm:ss a");
+    public static final String getFormattedReminderDate(Date date){
```
Comment (aayushKhandpur) [inline src/main/java/util/LocalizationUtil.java]: Util should not have methods specific to any other class as it is intended to be used by any class/method.

**Review by aayushKhandpur — CHANGES_REQUESTED** [review]
Address reviews

File: src/main/java/creativei/service/impl/ReminderServiceImpl.java [hunk 3]
```diff
@@ -1,7 +1,40 @@
...
 @Service("ReminderService")
-public class ReminderServiceImpl {
+public class ReminderServiceImpl implements ReminderService{
+    @Autowired
+    ReminderCustomDao reminderCustomDao;
+
+    Logger logger= LoggerFactory.getLogger(ReminderServiceImpl.class);
+
+    @Override
+    public Reminder create(Reminder reminder) throws InvalidParamRequest {
```
Comment (aayushKhandpur) [inline src/main/java/creativei/service/impl/ReminderServiceImpl.java:22]: Add error logs when error

File: src/main/java/creativei/service/impl/ReminderServiceImpl.java [hunk 4]
```diff
@@ -1,7 +1,40 @@
...
 import org.springframework.stereotype.Service;
+import util.StringUtil;
 
 @Service("ReminderService")
-public class ReminderServiceImpl {
+public class ReminderServiceImpl implements ReminderService{
+    @Autowired
+    ReminderCustomDao reminderCustomDao;
+
+    Logger logger= LoggerFactory.getLogger(ReminderServiceImpl.class);
```
Comment (aayushKhandpur) [inline src/main/java/creativei/service/impl/ReminderServiceImpl.java:19]: Use logger for both info and error logs in the methods

**Review by aayushKhandpur — CHANGES_REQUESTED** [review]
Address reviews

File: src/main/java/util/LocalizationUtil.java [hunk 5]
```diff
@@ -16,4 +16,14 @@ public static final Date stringToDateConverter(String string) throws ParseExcept
             return null;
         return dateFormat.parse(string);
     }
+    static SimpleDateFormat simpleDateFormat=new SimpleDateFormat("yyyy-MM-dd hh:mm:ss a");
```
Comment (aayushKhandpur) [inline src/main/java/util/LocalizationUtil.java]: We don't need seconds in the time

File: src/main/java/creativei/entity/Reminder.java (hunk 1 above)
Comment (aayushKhandpur) [reply src/main/java/creativei/entity/Reminder.java:18]: Make sure this is an optional field

File: src/main/java/creativei/service/impl/ReminderServiceImpl.java (hunk 4 above)
Comment (jitendrajaria) [reply src/main/java/creativei/service/impl/ReminderServiceImpl.java:19]: resolved in commit 18a0688

File: src/main/java/creativei/service/impl/ReminderServiceImpl.java (hunk 3 above)
Comment (jitendrajaria) [reply src/main/java/creativei/service/impl/ReminderServiceImpl.java:22]: resolved in commit   18a0688

File: src/main/java/creativei/entity/Reminder.java (hunk 1 above)
Comment (jitendrajaria) [reply src/main/java/creativei/entity/Reminder.java:18]: It is an optional field.

File: src/main/java/util/LocalizationUtil.java (hunk 5 above)
Comment (jitendrajaria) [reply src/main/java/util/LocalizationUtil.java]: issue resolved in commit 18a0688

File: src/main/java/util/LocalizationUtil.java (hunk 2 above)
Comment (jitendrajaria) [reply src/main/java/util/LocalizationUtil.java]: issue resolved in commit 18a0688

**Review by aayushKhandpur — CHANGES_REQUESTED** [review]
Address review

**Review by aayushKhandpur — CHANGES_REQUESTED** [review]
Address review

File: src/main/java/util/LocalizationUtil.java [hunk 6]
```diff
@@ -6,12 +6,13 @@
 
 public class LocalizationUtil {
 
-    static SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
-    public static final String getFormattedDate(Date date){
+    public static final String getFormattedDate(Date date,String stringDate){
```
Comment (aayushKhandpur) [inline src/main/java/util/LocalizationUtil.java]: Passing string format to the method is not the correct way as this util tends to lose the whole point. This util should have multiple methods based on requirements. And private static final string formats as class members.

Example class  member:
Private static final String DATE_WITH_TIME = "";
Example methods:

getDatFormatWithTime(Date date)

Comment (aayushKhandpur) [inline src/main/java/util/LocalizationUtil.java]: Passing string format to the method is not the correct way as this util tends to lose the whole point. This util should have multiple methods based on requirements. And private static final string formats as class members.

Example class  member:
Private static final String DATE_WITH_TIME = "";
Example methods:

getDatFormatWithTime(Date date)

Comment (jitendrajaria) [reply src/main/java/util/LocalizationUtil.java]: please review in commit bc8a190

**Review by aayushKhandpur — CHANGES_REQUESTED** [review]
Improve logs

File: src/main/java/creativei/service/impl/ReminderServiceImpl.java [hunk 7]
```diff
@@ -1,7 +1,42 @@
...
+    }
+
+    private Boolean validateReminder(Reminder reminder){
+        logger.info("Validate Reminder");
+        if(StringUtil.validateEmpty(reminder.getTitle())==null){
+            logger.error("Reminder title can not be null");
+            return false;
+        }
+        if(reminder.getReminderTime()==null){
+            logger.error("Reminder time can not be null");
```
Comment (aayushKhandpur) [inline src/main/java/creativei/service/impl/ReminderServiceImpl.java]: This is a log message. Log messages are for the application admins. Should say "reminder time is invalid: " + value

Comment (jitendrajaria) [reply src/main/java/creativei/service/impl/ReminderServiceImpl.java]: logger messages updated in commit 9c042d9

**Review by aayushKhandpur — APPROVED** [review]

**Pull request merged**